src/components/v2/appDetails/sourceInfo/environmentStatus/AppStatusDetailsChart.tsx

```diff
@@ -0,0 +1,144 @@
+        if (nodeStatusMap && nodeStatusMap.has(`${kind}/${name}`)) {
+            const { message } = nodeStatusMap.get(`${kind}/${name}`)
+            return message
+        }
+        return ''
+    }
+
+    const onFilterClick = (selectedFilter: string): void => {
+        if (currentFilter !== selectedFilter.toLowerCase()) {
+            setCurrentFilter(selectedFilter.toLowerCase())
+        }
+    }
+
+    const filteredFlattendNodes = () => {
+        if (filterRemoveHealth) {
+            return flattenedNodes?.filter((node) => node.health.status?.toLowerCase() !== DEPLOYMENT_STATUS.HEALTHY) || []
+        } else {
+            return flattenedNodes || []
+        }
+    }
+
+    return (
+        <div className="pb-12">
+            {filteredFlattendNodes().length > 0 && (
+                <div className="pt-16 pl-20 pb-8">
+                    <div className="flexbox pr-20 w-100">
+                        <div>
+                            <StatusFilterButtonComponent
+                                nodes={filteredFlattendNodes()}
+                                handleFilterClick={onFilterClick}
+                            />
+                        </div>
+                    </div>
+                </div>
+            )}
+            <div>
+                <div className="app-status-row dc__border-bottom pt-8 pr-20 pb-8 pl-20">
+                    {APP_STATUS_HEADERS.map((headerKey, index) => (
+                        <div className="fs-13 fw-6 cn-7" key={`header_${index}`}>
+                            {headerKey}
+                        </div>
+                    ))}
+                </div>
+                <div className={`resource-list fs-13 ${showFooter ? 'with-footer' : ''}`}>
+                    {filteredFlattendNodes().length > 0 ? (
+                        filteredFlattendNodes()
+                            .filter(
+                                (nodeDetails) =>
+                                    currentFilter === 'all' ||
+                                    nodeDetails.health.status?.toLowerCase() === currentFilter,
+                            )
+                            .map((nodeDetails) => (
+                                <div
+                                    className="app-status-row pt-8 pr-20 pb-8 pl-20"
+                                    key={`${nodeDetails.kind}/${nodeDetails.name}`}
+                                >
+                                    <div>{nodeDetails.kind}</div>
+                                    <div>{nodeDetails.name}</div>
+                                    <div
+                                        className={`app-summary__status-name f-${
+                                            nodeDetails.health.status ? nodeDetails.health.status.toLowerCase() : ''
+                                        }`}
+                                    >
+                                        {nodeDetails.status ? nodeDetails.status : nodeDetails.health.status}
```

Comment:
why 2 diff checks?

Reply:
because we are showing health status if node details status is not available